zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1128,9 +1135,24 @@ public void onSuccess(Note note, ServiceContext context) throws IOException {
         });
   }
 
+  protected void convertNote(NotebookSocket conn, Message fromMessage) throws IOException {
+    String note = gson.toJson(fromMessage.get("note"));
+
+    Message resp = new Message(OP.CONVERT_NOTE_NBFORMAT)
+            .put("nbformat", new JupyterUtil().getNbformat(note))
+            .put("name", fromMessage.get("name"));
+
+    conn.send(serializeMessage(resp));
+  }
+          
   protected Note importNote(NotebookSocket conn, Message fromMessage) throws IOException {
+    String noteJson;
     String noteName = (String) ((Map) fromMessage.get("note")).get("name");
-    String noteJson = gson.toJson(fromMessage.get("note"));
+    if (((Map) fromMessage.get("note")).get("cells") == null) {
```

Comment:
Add one line of comment to explain how we determine whether it is zeppelin note or jupyter note